pkg/cmd/cli/cmd/edit.go

```diff
@@ -412,3 +428,15 @@ func hasLines(r io.Reader) (bool, error) {
 	}
 	return false, nil
 }
+
+// rmComments removes the comments from a given byte slice
+func rmComments(content []byte) (raw []byte) {
+	lines := bytes.SplitAfter(content, []byte("\n"))
```

Comment:
We already have a function somewhere to do this - check the yaml package (I think).  This doesn't handle lines that have indented comments either.